src/nvim/autocmd.c

```diff
@@ -2005,6 +2005,52 @@ void auto_next_pat(AutoPatCmd *apc, int stop_at_last)
   }
 }
 
+static bool call_autocmd_callback(const AutoCmd *ac, const AutoPatCmd *apc)
+{
+  bool ret = false;
+  Callback callback = ac->exec.callable.cb;
+  if (callback.type == kCallbackLua) {
+    Dictionary data = ARRAY_DICT_INIT;
+    PUT(data, "match", CSTR_TO_OBJ((char *)autocmd_match));
```

Comment:
I think match is a fundamental property of the fired event, and weird to have stuffed away in a nested `"data"` dict

Reply:
.. which doesn't effect what I just said in the slightest? Please comment if you have anything useful to add.